api/topic.go

```diff
@@ -19,18 +19,51 @@ func (api *API) GetTopicHandler(w http.ResponseWriter, req *http.Request) {
 		handlers.CollectionID.Header(): hColID,
 		"request_id":                   ctx.Value(dprequest.RequestIdKey),
 		"topic_id":                     id,
+		"function":                     "getTopicHandler",
 	}
 
 	// get topic from mongoDB by id
-	topic, err := api.mongoDB.GetTopic(req.Context(), id)
+	topic, err := api.dataStore.Backend.GetTopic(req.Context(), id)
 	if err != nil {
 		handleError(ctx, w, err, logdata)
 		return
 	}
 
-	if err := WriteJSONBody(ctx, topic, w, logdata); err != nil {
+	// Ensure the sub document has the main document ID
+	topic.Current.ID = topic.ID
+
+	// User is not authenticated and hence has only access to current sub document
+	if err := WriteJSONBody(ctx, topic.Current, w, logdata); err != nil {
+		return
+	}
+	log.Event(ctx, "request successful", log.INFO, logdata) //!!! is this a good log to have ? ... as in is there too much logging going on ?
```

Comment:
"request successful" doesn't give any context - maybe `"get topic request completed successfully"` feels more descriptive and useful.

Reply:
all thes logs of success are like pouring fuel on the flames of the HEAP and garbage collector and are the achilies heal of microservices ...
The name of the function that just returned success is in this:
	logdata := log.Data{
		handlers.CollectionID.Header(): hColID,
		"request_id":                   ctx.Value(dprequest.RequestIdKey),
		"topic_id":                     id,
		"function":                     "getTopicHandler",
	}

Which is also passed into handleError() & WriteJSONBody() to show what failed without wasting HEAP space with a stack trace.

So no change needed, but i have added a comment to point this out.
Thanks.